andrada12/assignment4/WorldMap.java

```diff
@@ -0,0 +1,135 @@
+
+import java.util.ArrayList;
+import java.util.TreeSet;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author andrada
+ */
+
+ class MapPart{
+        int islandNumber;
+        char type;
```

Comment:
Rather than have a char which is 'T' or not, why not use a boolean?

Another interesting thing you could do is add two methods:
static MapPart land(int number);
static MapPart water(int number); which would make the creation of these much easier.